app/directory.rb

```diff
@@ -2,46 +2,59 @@
 
 module Tint
 	class Directory
-		def initialize(path)
-			@path = Pathname.new(path).realpath.cleanpath
-			unless @path.to_s.start_with?(PROJECT_PATH.to_s)
-				raise "File is outside of project scope!"
-			end
+		attr_reader :relative_path
+
+		def initialize(site, relative_path)
+			@site = site
+			@relative_path = Pathname.new(relative_path).cleanpath
+		end
+
+		def user_id
+			@site.user_id
```

Comment:
I generally prefer to make an `attr_reader` for the instance variable in case we modify something about how it is returned later. If we want it to be protected, we can define `attr_reader :site` after the `protected` keyword. Thoughts?